src/include/txn.i

```diff
@@ -218,37 +222,27 @@ __wt_txn_resolve_prepared_op(
 	    (WT_CURSOR_BTREE *)cursor, &upd));
 	WT_ERR(ret);
 
-#ifdef HAVE_DIAGNOSTIC
-	/*
-	 * Do what we can to ensure that finding prepared updates from a key
-	 * is working as expected. In the case where a transaction has updated
-	 * the same key multiple times, it's possible to resolve all updates
-	 * for the key when processing the first op structure, and then have
-	 * eviction free those updates before subsequent ops are processed,
-	 * which means a search could reasonably not find an update in that
-	 * case.
-	 * We track the update count only for commit, but not for rollback, as
-	 * our tracking is based on transaction id, and in case of rollback, we
-	 * set it to aborted.
-	 */
-	if (upd == NULL && commit) {
-		/*
-		 * FIXME:
-		 * WT_ASSERT(session, txn->multi_update_count > 0);
-		 */
-		--txn->multi_update_count;
+	/* If we haven't found anything then there's an error. */
+	if (upd == NULL) {
+		WT_ASSERT(session, upd != NULL);
+		WT_ERR(WT_NOTFOUND);
 	}
-#endif
-
-	WT_STAT_CONN_INCR(session, txn_prepared_updates_resolved);
 
 	for (; upd != NULL; upd = upd->next) {
-		 if (upd->txnid != txn->id)
-			continue;
-
+		/*
+		 * Aborted updates can exist in the update chain of our TXN
+		 * generally this will occur due to a reserved update.
```

Comment:
Minor nits: 'TXN' should lowercase. Also, the first line is a complete sentence. Please end it with a period and capitalize the 'G' in 'generally' on the next line.